datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnersModal.tsx

```diff
@@ -0,0 +1,294 @@
+import { OWNERSHIP_DISPLAY_TYPES } from './ownershipUtils';
+import { useAddOwnersMutation } from '../../../../../../../graphql/mutations.generated';
+import { useGetSearchResultsLazyQuery } from '../../../../../../../graphql/search.generated';
+import OwnerPill from '../../../../../../shared/OwnerPill';
+
+const SearchResultContainer = styled.div`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    padding: 2px;
+`;
+
+const SearchResultContent = styled.div`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+`;
+
+const SelectInput = styled(Select)`
+    > .ant-select-selector {
+        height: 36px;
+    }
+`;
+
+const SearchResultDisplayName = styled.div``;
+
+type Props = {
+    urn: string;
+    type: EntityType;
+    visible: boolean;
+    defaultOwnerType?: OwnershipType;
+    hideOwnerType?: boolean | undefined;
+    onCloseModal: () => void;
+    refetch?: () => Promise<any>;
+};
+
+type SelectedOwner = {
+    label: string;
+    value: any;
+};
+
+export const AddOwnersModal = ({
+    urn,
+    type,
+    visible,
+    hideOwnerType,
+    defaultOwnerType,
+    onCloseModal,
+    refetch,
+}: Props) => {
+    const entityRegistry = useEntityRegistry();
+    const [addOwnersMutation] = useAddOwnersMutation();
+    const ownershipTypes = OWNERSHIP_DISPLAY_TYPES;
+    const [selectedOwners, setSelectedOwners] = useState<SelectedOwner[]>([]);
+    const [selectedOwnerType, setSelectedOwnerType] = useState<OwnershipType>(defaultOwnerType || OwnershipType.None);
+
+    // User and group dropdown search results!
+    const [userSearch, { data: userSearchData }] = useGetSearchResultsLazyQuery();
+    const [groupSearch, { data: groupSearchData }] = useGetSearchResultsLazyQuery();
+    const userSearchResults = userSearchData?.search?.searchResults || [];
+    const groupSearchResults = groupSearchData?.search?.searchResults || [];
+    const combinedSearchResults = [...userSearchResults, ...groupSearchResults];
+
+    // Add owners Form
+    const [form] = Form.useForm();
+
+    useEffect(() => {
+        if (ownershipTypes) {
+            setSelectedOwnerType(ownershipTypes[0].type);
+        }
+    }, [ownershipTypes]);
+
+    // When a owner search result is selected, add the new owner  to the selectedOwners
+    const onSelectOwner = (newSelectedOwner) => {
+        const filteredActors = combinedSearchResults
+            .filter((result) => result.entity.urn === newSelectedOwner.value)
+            .map((result) => result.entity);
+        if (filteredActors.length) {
+            const actor = filteredActors[0];
+            const ownerEntityType =
+                actor && actor.type === EntityType.CorpGroup ? OwnerEntityType.CorpGroup : OwnerEntityType.CorpUser;
+            const newValues = [
+                ...selectedOwners,
+                {
+                    label: newSelectedOwner.value,
```

Comment:
Let's add a comment to say that the "value" field will be the urn of the owner.

Reply:
Added and committed

Reply:
Added and Committed